src/input/apple/InputAppleMediaKeys.mm

```diff
@@ -69,9 +122,97 @@ -(void)mediaKeyTap:(SPMediaKeyTap *)keyTap receivedMediaKeyEvent:(NSEvent *)even
 
 @end
 
+// macOS private enum
+enum {
+    MRNowPlayingClientVisibilityUndefined = 0,
+    MRNowPlayingClientVisibilityAlwaysVisible,
+    MRNowPlayingClientVisibilityVisibleWhenBackgrounded,
+    MRNowPlayingClientVisibilityNeverVisible
+};
+
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 bool InputAppleMediaKeys::initInput()
 {
+  m_currentTime = 0;
+  m_pendingUpdate = false;
   m_delegate = [[MediaKeysDelegate alloc] initWithInput:this];
+  if (NSClassFromString(@"MPNowPlayingInfoCenter")) {
+    connect(&PlayerComponent::Get(), &PlayerComponent::stateChanged, this, &InputAppleMediaKeys::handleStateChanged);
+    connect(&PlayerComponent::Get(), &PlayerComponent::positionUpdate, this, &InputAppleMediaKeys::handlePositionUpdate);
+    connect(&PlayerComponent::Get(), &PlayerComponent::updateDuration, this, &InputAppleMediaKeys::handleUpdateDuration);
+    void* lib = dlopen("/System/Library/PrivateFrameworks/MediaRemote.framework/MediaRemote", RTLD_NOW);
+    if (lib) {
+#define LOAD_FUNC(name) \
+  name = (name ## Func)dlsym(lib, "MRMediaRemote" #name)
+      LOAD_FUNC(SetNowPlayingVisibility);
+      LOAD_FUNC(GetLocalOrigin);
+      LOAD_FUNC(SetCanBeNowPlayingApplication);
+      if (SetCanBeNowPlayingApplication)
+        SetCanBeNowPlayingApplication(1);
+    }
```

Comment:
I suppose these are all private API functions. This looks OK - if Apple leaves us no choice, it's probably just Apple being Apple, and we can gracefully handle the unlikely event of Apple removing or hiding these functions.